### ISSUE TITLE
F-Droid inclusion

### ISSUE THREAD

**Issue by shuvashish76** [httptoolkit/httptoolkit-android#12]
Would you consider publishing this app on F-Droid please?

According to Izzy [scanner report ](https://gitlab.com/IzzyOnDroid/repo/-/issues/387#note_1408304037):
```
Offending libs:
---------------
* Play Install Referrer Library (/com/android/installreferrer): NonFreeDep,NonFreeNet,Tracking
* Android Market (/com/google/android/finsky): NonFreeNet
* Google Mobile Services (/com/google/android/gms): NonFreeDep
* Firebase (/com/google/firebase): NonFreeNet,NonFreeDep
* Sentry SDK for Java (/io/sentry): Tracking

5 offenders.
```

As Izzy pointed out there, any possibility of keeping FOSS flavour without these libraries for F-Droid?

**Comment by pimterry** [comment]
Hi @shuvashish76 - sure, I'd happily welcome PRs to support this if you're interested in that.

That said, I should note that this app isn't primarily distributed via the Play store anyway. In most cases, it's used the the ADB activation option in HTTP Toolkit, which directly installs the app via ADB itself. I agree it might be nice to have it explicitly available via F-Droid too, but you can use this app on devices without the Play store anyway.

**Comment by shuvashish76** [comment]
Sorry, I can't help since I'm not a dev. :(
But F-Droid maintainers will take care the publishing part if you could provide the FOSS flavour.

**Comment by cHAuHaNz** [comment]
> the app isn't generally distributed via app stores at all

@pimterry can you help me understand the usage of this then? Do we have a particular case when user is redirected to store?

https://github.com/httptoolkit/httptoolkit-android/blob/3feedb7a95322711da380db3d6a3d46bf136eb46/app/src/main/java/tech/httptoolkit/android/MainActivity.kt#L177-L187

**Comment by pimterry** [comment]
If you scan the QR code from the desktop with a normal QR code scanner, without the app already installed, it sends you to the play store to install it (in that scenario, we can't install the app via ADB). What I meant by "the app isn't generally distributed via app stores at all" really is that nobody _discovers_ the app via the play store - they just get sent there directly as a download source. Since we don't know what's on the user's device we couldn't use F-Droid or any other store for this, we basically just have to use the most popular store available.

It's not a very common flow really, most users use ADB setup, and I suspect when the QR code is used it's often used on devices that do already have the app installed anyway. I haven't actually tested the post-install referrer logic in years so tbh I'm not 100% sure it still works, but it's not actually critical - if it doesn't, you just end up inside the app and you scan the code again.

**Comment by cHAuHaNz** [comment]
@pimterry Well personally I also don't think its much of a critical feature as its a really an edge of edge case. As you said, mostly users go with setup via ADB which installs the app directly on the device, and even if someone wants to go out of way and scan the QR using a normal QR scanner, still they can scan the QR once again after they land on Home screen of app. Let me know if you are fine with the removal of libraries stated by OP, I can make the changes and raise a PR for the same.

**Comment by pimterry** [comment]
@cHAuHaNz Thanks for the offer, but I don't think this is worthwhile. I'll close the issue. I think in most cases this app will be sideloaded via ADB, and in the few other cases you often want Google Play installed anyway to install 3rd party apps you're interested in intercepting, so the use cases for f-droid distribution are very very limited.

We could remove the libraries listed above without breaking core functionality, but they are actively mildly useful (e.g. Sentry provides error reporting, Play libs are used to detect & prompt for app updates on-device, referrer supports the use case described above) and since there's nearly no benefit, I don't think we should.

**Comment by cHAuHaNz** [comment]
Understood 😄
